### PR TITLE
CCP-1265: Review commit

### PR DESCRIPTION
**Pull request by dqdquint**
Left default values in order for testing. There still seems to be an issue with Android devices opening these links but discussions led me to believe this is a version issue with our android test devices. May require additional investigation. 

Tested with shop number 31741. Product widgets on home page have items with and without download property.

Unit test requiring mock store is blocked by: https://jira.shopgate.guru/browse/PWA-1676

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## [1.0.0] - 02-20-2019
+First version of the extension.
+### Added
+- Added support to display pdf files passed through properties on pdp pages.
+- Configurations include sectionTitle, manualsProperty, manualsPrefixUrl, typeIndices, and propertySplit.
```
Comment (sznowicki) [inline CHANGELOG.md]: Link to diff is missing. Please also tag initial commit with v0.0.1 so there is something to diff

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## [1.0.0] - 02-20-2019
```
Comment (sznowicki) [inline CHANGELOG.md]: Please use ISO date format: 2019-12-31

File: README.md [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+      {
+        "searchPattern": "B.pdf",
+        "fileType": "Brochure"
+      },
+      {
+        "searchPattern": "D.pdf",
+        "fileType": "Documentation"
+      }
+    ],
+  "manualsPrefixUrl": "https://www.shopgate.com/productdownloads/",
```
Comment (sznowicki) [inline README.md]: It's 404. Please change it to https://example.com/examplePathname

File: extension-config.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      "params": {
+        "required": true,
+        "type": "text",
+        "label": "Name of product property which contains manuals urls"
+      }
+    },
+    "manualsPrefixUrl": {
+      "type": "admin",
+      "destination": "frontend",
+      "default": "https://www.pollin.de/productdownloads/",
```
Comment (sznowicki) [inline extension-config.json]: This must not be here

**Review by sznowicki — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## [1.0.0] - 2019-02-20
+First version of the extension.
+### Added
+- Added support to display pdf files passed through properties on pdp pages.
+- Configurations include sectionTitle, manualsProperty, manualsPrefixUrl, typeIndices, and propertySplit.
+
+[Link to Diff](https://github.com/shopgate-professional-services/ext-pdp-manual-section/compare/v0.0.1...v1.0.0)
```
Comment (sznowicki) [inline CHANGELOG.md]: Please adjust the changelog to our format.

File: CHANGELOG.md (hunk 1 above)
Comment (sznowicki) [reply CHANGELOG.md]: Change is not correct. Also please respond in the pull request comments

File: README.md (hunk 3 above)
Comment (sznowicki) [reply README.md]: done

File: CHANGELOG.md (hunk 2 above)
Comment (sznowicki) [reply CHANGELOG.md]: done

File: extension-config.json (hunk 4 above)
Comment (sznowicki) [reply extension-config.json]: done

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# To use .editorconfig you need to install a plugin from
```
Comment (sznowicki) [inline .editorconfig:1]: Frontend tests are failing. I'm stop reviewing at this point.

**Review by sznowicki — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 6 above)
Comment (dqdquint) [reply .editorconfig:1]: Updated failed snapshot

File: CHANGELOG.md (hunk 1 above)
Comment (dqdquint) [reply CHANGELOG.md]: Updated link to diff. Deleted added tag for v1.0.0. Shouldn't have been added yet.

File: .editorconfig (hunk 6 above)
Comment (sznowicki) [reply .editorconfig:1]: One sure, the other is still wrong. Please make sure tests are green before putting this into review.

**Review by sznowicki — CHANGES_REQUESTED** [review]
Tests are still failing. Stop reviewing at this point.

File: CHANGELOG.md (hunk 5 above)
Comment (dqdquint) [reply CHANGELOG.md]: Updated to reflect format. Will add tag when approved

File: .editorconfig (hunk 6 above)
Comment (dqdquint) [reply .editorconfig:1]: Updated tests to reflect config file with empty object

Comment (sznowicki) [reply .editorconfig:1]: @DannyShopgate what you did in last commit is using empty config indeed but now you just also updated the snapshot so it tests against component that returns null.

We talked about it and I told you you need to mock the config in order to test if it renderers. Testing component that doesn't render anything doesn't make much sense.

Please adjust this test.

**Review by sznowicki — CHANGES_REQUESTED** [review]

File: frontend/selectors/index.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  getProductPropertiesUnfiltered,
+  (properties) => {
+    const parsedManualUrls = [];
+    if (!properties) {
+      return parsedManualUrls;
+    }
+    const manualUrls = properties.find(property => (
+      property.label === manualsProperty
+    ));
+    if (manualUrls) {
```
Comment (sznowicki) [inline frontend/selectors/index.js]: Please return early instead of wrapping entire, last block in the condition.

File: frontend/selectors/index.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      manualUrls.value.split(propertySplit).forEach((url) => {
+        const filteredObject = typeIndices.filter((index) => {
+          const regex = new RegExp(index.searchPattern, 'gi');
+          return regex.test(url);
+        });
+        if (filteredObject.length > 0) {
```
Comment (sznowicki) [inline frontend/selectors/index.js]: Return early as above.

File: frontend/components/ManualsSection/index.jsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    <div className={styles.spacer}>
+      <div className={styles.container}>
+        <span className={styles.headline}>{config.sectionTitle}</span>
+        {manualUrls.map(value => ((
+          <Link
+            key={value.label}
+            className={styles.singleLink}
+            href={value.url}
+          >
+            <FileIcon className={styles.icon} />&nbsp;&nbsp; Open {value.type}
```
Comment (sznowicki) [inline frontend/components/ManualsSection/index.jsx]: Tip for  the future: try to review your own pull requests before submitting it. You would find this kind of mistakes easily.

File: frontend/Portals/ProductDescriptionAfter/index.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  <ManualsSection
+    config={config}
+    manualUrls={manualUrls}
+  />
+));
+/**
+ * @returns {JSX}
+ */
+const ProductDescriptionAfter = () => {
+  if (config.manualsProperty) {
```
Comment (sznowicki) [inline frontend/Portals/ProductDescriptionAfter/index.jsx]: Please return early.

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
+language: node_js
+
+install:
+- cd extension; npm install; cd ..
```
Comment (sznowicki) [inline .travis.yml]: There's no way this travis-ci would work.

Comment (sznowicki) [reply .travis.yml]: Please make this config work. I'd enable it for you, right after I get permissions to this repo.

Comment (sznowicki) [reply .travis.yml]: Also enable coverage, you can find examples e.x. on pwa-extension-kit.

Comment (sznowicki) [reply .travis.yml]: I cannot enable coverage, since I have no sufficient rights to this repo. Please follow our confluence page regarding how to create github repo.

**Review by sznowicki — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 11 above)
Comment (dqdquint) [reply .travis.yml]: Enabled coverage and configured .travis.yml file and confirmed coveralls is enabled for this repo.

File: frontend/components/ManualsSection/index.jsx (hunk 9 above)
Comment (dqdquint) [reply frontend/components/ManualsSection/index.jsx]: image
Hey @sznowicki, I'm not sure which mistake you are referencing here. The image is the file icon and the value added via config

File: frontend/Portals/ProductDescriptionAfter/index.jsx (hunk 10 above)
Comment (dqdquint) [reply frontend/Portals/ProductDescriptionAfter/index.jsx]: Updated

File: frontend/selectors/index.js (hunk 8 above)
Comment (dqdquint) [reply frontend/selectors/index.js]: Updated

File: frontend/selectors/index.js (hunk 7 above)
Comment (dqdquint) [reply frontend/selectors/index.js]: Updated

File: .editorconfig (hunk 6 above)
Comment (dqdquint) [reply .editorconfig:1]: Heyo @sznowicki I believe this is the test in questions: 
```javascript
const noManualUrls = null;
const manualUrls = [{
  label: 'mock',
  url: 'mock',
  type: 'mock',
}];
const mockConfig = { sectionTitle: 'mock' };
describe('<ManualsSection />', () => {
  it('should not render if no manual urls', () => {
    const wrapper = mount((
      <ManualsSection config={mockConfig} manualUrls={noManualUrls} />
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').exists()).toBe(false);
  });
  it('should render if provided with manual urls', () => {
    const wrapper = mount((
      <ManualsSection config={mockConfig} manualUrls={manualUrls} />
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').exists()).toBe(true);
  });
});
```
I i did mock a config passed to the component. The second test, tests if the component does render.

**Comment by dqdquint** [comment]
Heyo @sznowicki  I configured travis and coveralls but I had to some review questions. 
Also just fyi I tested with shop no. 31741 and used example configuration in Readme as config for testing. An example item no. is 630845

File: frontend/components/ManualsSection/index.jsx (hunk 9 above)
Comment (sznowicki) [reply frontend/components/ManualsSection/index.jsx]: There is hardcoded "Open" language string. This extension should be localised.

File: .travis.yml (hunk 11 above)
Comment (sznowicki) [reply .travis.yml]: Coveralls is enabled but travis fails to send the data to coveralls:
```
0.00s$ cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js
/home/travis/.travis/functions: line 104: ./node_modules/coveralls/bin/coveralls.js: No such file or directory
```
I'm pretty sure coveralls npm package is missing.

File: frontend/portals/ProductDescriptionAfter/__snapshots__/index.spec.jsx.snap [hunk 12]
```diff
@@ -0,0 +1,20 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (sznowicki) [inline frontend/portals/ProductDescriptionAfter/__snapshots__/index.spec.jsx.snap:1]: Could you please explain me why did you renamed the Portals directory to "portals"?

I see this is only renamed here, but it's not changed in the extension config. Which means it would fail during the deployment on any case-sensitive machine.

File: .editorconfig (hunk 6 above)
Comment (sznowicki) [reply .editorconfig:1]: No, we're still talking about the ProductDescriptionAfter, the one that was failing before because of missing config mock.

Not it still tests against "live" config which is an empty file atm. It must be mocked, otherwise the test renders empty component, which doesn't make any sense for testing.

**Review by sznowicki — CHANGES_REQUESTED** [review]

File: frontend/components/ManualsSection/index.jsx (hunk 9 above)
Comment (dqdquint) [reply frontend/components/ManualsSection/index.jsx]: Gotcha. Updated

File: .editorconfig (hunk 6 above)
Comment (dqdquint) [reply .editorconfig:1]: Right, of course! Updated

File: .travis.yml (hunk 11 above)
Comment (dqdquint) [reply .travis.yml]: Added coveralls

File: frontend/portals/ProductDescriptionAfter/__snapshots__/index.spec.jsx.snap (hunk 12 above)
Comment (dqdquint) [reply frontend/portals/ProductDescriptionAfter/__snapshots__/index.spec.jsx.snap:1]: I updated directory name to reflect other directory casing. Discovered an issue in travis with unit tests failing because directory could not be found. Tests were not failing on my machine because it is not case sensitive. Updated extension config directory name

**Review by sznowicki — APPROVED** [review]
